File: Daily Challanges/2026-04-April/2026-04-02.py
def capitalize_fibonacci(s):
    n = len(s)
    fib = set()
    a, b = 0, 1
    while a < n:
        fib.add(a)
        a, b = b, a + b

    res = []
    for i, c in enumerate(s):
        if c.isalpha():
            if i in fib:
                res.append(c.upper())
            else:
                res.append(c.lower())
        else:
            res.append(c)
    return "".join(res)

File: Daily Challanges/2026-04-April/2026-04-02.js
function capitalizeFibonacci(s) {
  const n = s.length;
  const fib = new Set();
  let a = 0,
    b = 1;
  while (a < n) {
    fib.add(a);
    [a, b] = [b, a + b];
  }
  let res = "";
  for (let i = 0; i < n; i++) {
    const c = s[i];
    if (/[a-zA-Z]/.test(c)) {
      if (fib.has(i)) res += c.toUpperCase();
      else res += c.toLowerCase();
    } else {
      res += c;
    }
  }
  return res;
}
